### PR TITLE
lightning: support detect duplicate keys on specific index ID

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref #35983

Problem Summary:

To support adding index through lightning, an essential feature is to detect the duplicate key for the unique indexes.

### What is changed and how it works?

This PR adds support for `Backend` to collect duplicate keys on a specific index ID.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/backend/local/duplicate.go [hunk 1]
```diff
@@ -579,6 +584,29 @@ func (m *DuplicateManager) buildDupTasks() ([]dupTask, error) {
 	return tasks, nil
 }
 
+func (m *DuplicateManager) buildIndexDupTasks() ([]dupTask, error) {
```
Comment (gozssky) [inline br/pkg/lightning/backend/local/duplicate.go:587]: Could you please add a unit test for this function? (I'm so sorry there is no test for this file before)

**Review by gozssky — APPROVED** [review]

File: br/pkg/lightning/backend/local/duplicate.go (hunk 1 above)
Comment (tangenta) [reply br/pkg/lightning/backend/local/duplicate.go:592]: Done.

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**